vector-stores/spring-ai-cassandra/src/main/java/org/springframework/ai/vectorstore/CassandraVectorStore.java

```diff
@@ -0,0 +1,412 @@
+
+	private final CassandraVectorStoreConfig conf;
+
+	private final EmbeddingClient embeddingClient;
+
+	private FilterExpressionConverter filterExpressionConverter;
+
+	private final PreparedStatement addStmt;
+
+	private final PreparedStatement deleteStmt;
+
+	private final PreparedStatement similarityStmt;
+
+	private final Similarity similarity;
+
+	public CassandraVectorStore(CassandraVectorStoreConfig conf, EmbeddingClient embeddingClient) {
+
+		Assert.notNull(conf, "Config must not be null");
+		Assert.notNull(embeddingClient, "Embedding client must not be null");
+
+		this.conf = conf;
+		this.embeddingClient = embeddingClient;
+		this.filterExpressionConverter = null; // TODO new
+												// CassandraFilterExpressionConverter(this.config.metadataFields);
+		ensureSchemaExists(conf, embeddingClient.dimensions());
+
+		addStmt = conf.session.prepare(QueryBuilder.insertInto(conf.keyspace, conf.table)
+			.value(conf.id, QueryBuilder.bindMarker("idName"))
+			.value(conf.content, QueryBuilder.bindMarker("contentFieldName"))
+			.value(conf.embedding, QueryBuilder.bindMarker("embeddingFieldName"))
+			.build());
+
+		deleteStmt = conf.session.prepare(QueryBuilder.deleteFrom(conf.keyspace, conf.table)
+			.whereColumn(conf.id)
+			.isEqualTo(QueryBuilder.bindMarker("idName"))
+			.build());
+
+		similarity = Similarity.valueOf(conf.session.getMetadata()
+			.getKeyspace(conf.keyspace)
+			.get()
+			.getTable(conf.table)
+			.get()
+			.getIndex(conf.index)
+			.get()
+			.getOptions()
+			.getOrDefault("similarity_function", "COSINE")
+			.toUpperCase());
+
+		String similarityFunc = format("similarity_%s(%s,?)", similarity.toString().toLowerCase(), conf.embedding);
+
+		// java-driver-query-builder doesn't support orderByAnnOf yet
+		String query = format(QUERY_FORMAT, conf.id, conf.content, similarityFunc, conf.keyspace, conf.table,
+				conf.embedding);
+		logger.debug("preparing {}", query);
+		similarityStmt = conf.session.prepare(query);
+	}
+
+	@Override
+	public void add(List<Document> documents) {
+		CompletableFuture[] futures = new CompletableFuture[documents.size()];
+		short i = 0;
+		for (Document d : documents) {
+			List<Float> embedding = embeddingClient.embed(d).stream().map(Double::floatValue).toList();
+			BoundStatement s = addStmt.bind(d.getId(), d.getContent(), CqlVector.newInstance(embedding));
+			futures[i++] = conf.session.executeAsync(s).toCompletableFuture();
+		}
+		CompletableFuture.allOf(futures).join();
+	}
+
+	@Override
+	public Optional<Boolean> delete(List<String> idList) {
+		CompletableFuture[] futures = new CompletableFuture[idList.size()];
+		short i = 0;
+		for (String id : idList) {
+			BoundStatement s = deleteStmt.bind(id);
+			futures[i++] = conf.session.executeAsync(s).toCompletableFuture();
+		}
+		CompletableFuture.allOf(futures).join();
+		return Optional.of(Boolean.TRUE);
+	}
+
+	@Override
+	public List<Document> similaritySearch(SearchRequest request) {
```

Comment:
Most of the VectorStore integrations provide support for the [Metadata Expression Filtering](https://docs.spring.io/spring-ai/reference/1.0-SNAPSHOT/api/vectordbs.html#metadata-filters).
Please consider adding filtering support. 
Otherwise, please throw an exception if someone set at filter to indicate that the feature is not supported [like this](https://github.com/spring-projects/spring-ai/blob/9236913696298bd42838bab0b72416ab413bf8d3/vector-stores/spring-ai-gemfire/src/main/java/org/springframework/ai/vectorstore/GemFireVectorStore.java#L465)

Reply:
it's coming.  (the PR is still in draft)